### DISCUSSION TITLE
Generating static page on backend with server function

### DISCUSSION THREAD

**Discussion by Raniz85** [leptos-rs/leptos#1269] [Q&A] [answered]
When running the todo_app_sqlite example locally and loading the page it is loaded with the fallback placeholder of the `<Transition>` element and then filled in with JavaScript on page load.

Is there any way to avoid this and serve the site completely pre-rendered instead so that it would work with no-script?

**Comment by gbj** [comment, marked as answer]
See [the docs on SSR rendering modes](https://leptos-rs.github.io/leptos/ssr/23_ssr_modes.html). `SsrMode::Async`, `SsrMode::InOrder`, or `SsrMode::PartiallyBlocked` (with `create_blocking_resource` for the todos list) all do what you're asking.

**Reply by Raniz85** [reply]
Yeah, I found that after I wrote the question. I looked under Progressive Enhancement and completely forgot to look under SSR :)
